packages/deployments/contracts/contracts/nomad-xapps/contracts/relayer-fee-router/RelayerFeeRouter.sol

```diff
@@ -0,0 +1,190 @@
+   * @dev Connext and relayer fee router store references to each other
+   * @param _connext The address of the Connext implementation
+   */
+  function setConnext(address _connext) external onlyOwner {
+    connext = IConnext(_connext);
+    emit SetConnext(_connext);
+  }
+
+  // ======== External: Send Claim =========
+
+  /**
+   * @notice Sends a request to claim the fees in the originated domain
+   * @param _domain The domain where to claim the fees
+   * @param _recipient The address of the relayer
+   * @param _amount The amount of fees to claim
+   * @param _transactionIds A group of transaction ids to claim for fee bumps
+   */
+  function send(
+    uint32 _domain,
+    address _recipient,
+    uint256 _amount,
+    bytes32[] calldata _transactionIds
+  ) external onlyConnext {
+    if (_amount == 0 && _transactionIds.length == 0) revert RelayerFeeRouter__send_claimEmpty();
+    if (_recipient == address(0)) revert RelayerFeeRouter__send_recipientEmpty();
+
+    // get remote RelayerFeeRouter address; revert if not found
+    bytes32 remote = _mustHaveRemote(_domain);
+
+    bytes memory message = RelayerFeeMessage.formatClaimFees(_recipient, _amount, _transactionIds);
+
+    xAppConnectionManager.home().dispatch(_domain, remote, message);
+
+    // emit Send event
+    emit Send(_domain, _recipient, _amount, _transactionIds, remote, message);
+  }
+
+  // ======== External: Handle =========
+
+  /**
+   * @notice Handles an incoming message
+   * @param _origin The origin domain
+   * @param _nonce The unique identifier for the message from origin to destination
+   * @param _sender The sender address
+   * @param _message The message
+   */
+  function handle(
+    uint32 _origin,
+    uint32 _nonce,
+    bytes32 _sender,
+    bytes memory _message
+  ) external override onlyReplica onlyRemoteRouter(_origin, _sender) {
+    // parse tokenId and action from message
+    bytes29 _msg = _message.ref(0).mustBeClaimFees();
+
+    address recipient = _msg.recipient();
+    uint256 amount = _msg.amount();
+    bytes32[] memory transactionIds = _msg.transactionIds();
+
+    // TODO - claim fee in connext
+
+    // emit Receive event
+    emit Receive(_originAndNonce(_origin, _nonce), _origin, recipient, amount, transactionIds);
+  }
+
+  /**
+   * @dev explicit override for compiler inheritance
+   * @dev explicit override for compiler inheritance
+   * @return domain of chain on which the contract is deployed
+   */
+  function _localDomain() internal view override(XAppConnectionClient) returns (uint32) {
+    return XAppConnectionClient._localDomain();
+  }
+
+  /**
+   * @notice Internal utility function that combines
+   * `_origin` and `_nonce`.
+   * @dev Both origin and nonce should be less than 2^32 - 1
+   * @param _origin Domain of chain where the transfer originated
+   * @param _nonce The unique identifier for the message from origin to destination
+   * @return Returns (`_origin` << 32) & `_nonce`
```

Comment:
nit: `@return` uses `AND &` where we use `OR |` below lol

Reply:
I think the return comment is designed to represent the semantic happenings not the actual operators used (i.e. returns left-shifted origin + nonce, not either or)

Reply:
oh lol